tools/windows_portable/shim/frame-compare-update.ps1

```
@@ -0,0 +1,739 @@
+    Ensure-Directory -Path (Split-Path -Parent $backupDir)
+    Copy-Item -LiteralPath $targetDir -Destination $backupDir -Recurse -Force
+
+    Copy-DirectoryContents -SourceDir $payloadDir -DestinationDir $newDir
+    Invoke-WithRetry -Label "Rename current frame_compare to .old" -Action { Move-Item -LiteralPath $targetDir -Destination $oldDir -Force }
+    Invoke-WithRetry -Label "Rename .new into place" -Action { Move-Item -LiteralPath $newDir -Destination $targetDir -Force }
+
+    try {
+      $targetVersion = Get-PayloadVersionFromManifest -Manifest $manifest
+      Invoke-SmokeCheck -BundlePath $BundlePath -ExpectedVersion $targetVersion
+    } catch {
+      Write-Host "Smoke check failed; rolling back update."
```

PSScriptAnalyzer (1.24.0)

[warning] File 'frame-compare-update.ps1' uses Write-Host. Avoid using Write-Host because it might not work in all hosts, does not work when there is no host, and (prior to PS 5.0) cannot be suppressed, captured, or redirected. Instead, use Write-Output, Write-Verbose, or Write-Information.

(PSAvoidUsingWriteHost)